### PR TITLE
data source and processing

### PR DESCRIPTION
**Pull request by uldrim**
1) Adding all the required files
2) The output of the demo algorithm is matching with the raw data (Attaching screenshots for the same)
3) The data being populated in the output/alternative/coingecko/marketcap directory is matching with the data available on Coingecko website (matched it by creating a GET request to coingecko API at /coins/{id}/history endpoint by specifying the particular dates and matching them)
btc
log2

### CODE REVIEW

**Comment by uldrim** [comment]
I have also added the data files for top 5 coins.
The marketcap for any particular date can be matched using the coingecko api endpoint '/coins/{id}/history'.
For example : 'https://api.coingecko.com/api/v3/coins/bitcoin/history?date=07-10-2013&localization=false'

File: CoinGeckoMarketCapAlgorithm.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        private Symbol _cryptoSymbol;
+
+        /// <summary>
+        /// Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.
+        /// </summary>
+        public override void Initialize()
+        {
+            SetStartDate(2013, 10, 07);  //Set Start Date
+            SetEndDate(2013, 10, 11);    //Set End Date
+            _cryptoSymbol = AddCrypto("BTCUSD").Symbol;
```
Comment (LouisSzeto) [inline CoinGeckoMarketCapAlgorithm.cs]: This line seems redundant as it is not used.

File: CoinGeckoMarketCapAlgorithm.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+### <summary>
+### Example algorithm using the custom data type as a source of alpha
+### </summary>
+class CoinGeckoMarketCapAlgorithm(QCAlgorithm):
+    def Initialize(self):
+        ''' Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.'''
+        
+        self.SetStartDate(2020, 10, 7)   #Set Start Date
+        self.SetEndDate(2020, 10, 11)    #Set End Date
+        self.crypto_symbol = self.AddEquity("BTCUSD", Resolution.Daily).Symbol
```
Comment (LouisSzeto) [inline CoinGeckoMarketCapAlgorithm.py]: Same comment.

File: process.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        return requests.get(f'{base_url}/{url}').json()
+
+
+if __name__ == "__main__":
+  start_time = time.time()
+  destinationDirectory = f"output/alternative/{VendorName}/{VendorDataName}"
+  instance = CoinGeckoMarketCapDataDownloader(destinationDirectory, db_file)
+  instance.Run()
+  time_taken = time.time() - start_time
+  print("Total time taken to run in minutes : ", time_taken//60)
```
Comment (LouisSzeto) [inline process.py]: Can you fix the nit here? 
Some of them use "Tab", some with "2 spaces" and some with "4 spaces". We should use "4 spaces" for all indents.

For all files: please remove any tab or space for the empty lines with no code.

**Review by LouisSzeto — APPROVED** [review]
Awesome! Just some minor comments.

File: listing-about.md [hunk 4]
```diff
@@ -1,45 +1,46 @@
...
 ### About the Provider
-*`vendorName`* was founded by *`founderName(s)`* in *`foundingYear`*, with the goal of *`missionStatement`*. *`vendorName`* provides access to *`dataTypes`* for *`targetDemographic`*. 
+CoinGecko provides a data platform, digital currency prices, and fundamental analysis of the cryptocurrency market.CoinGecko was founded in 2014 by TM Lee (CEO) and Bobby Ong (COO) with the mission to democratize the access of crypto data and empower users with actionable insights.
 
 ### Getting Started
 Python:
 ```
-# pythonCodeToSubscribeToData
+# Requesting data:
+self.btcusd = self.AddCrypto("BTCUSD", Resolution.Daily).Symbol
```
Comment (AlexCatarino) [inline listing-about.md]: Here we just want to teach how to add data of this dataset, so we don't need to add crypto. This is in line with Louis' comments above.

File: listing-about.md [hunk 5]
```diff
@@ -1,45 +1,46 @@
...
+coingecko_marketcap_symbol = self.AddData(CoinGeckoMarketCap, "BTC").Symbol
 ```
 
 C#:
 ```
-// cSharpCodeToSubscribeToData
+// Requesting data:
+btcusd = AddCrypto("BTCUSD", Resolution.Daily).Symbol;
```
Comment (AlexCatarino) [inline listing-about.md]: Same as previous.

File: listing-about.md [hunk 6]
```diff
@@ -1,45 +1,46 @@
...
 ### Data Summary
-- **Start Date**: *`startDate`*
-- **Asset Coverage**: *`assetCoverage`*
-- **Resolution**: *`resolution(s)`*
-- **Data Density**: *`dataDensity`*
-- **Timezone**: *`timezone`*
+- **Start Date**: April 2013
+- **Asset Coverage**: Crypto Coins
+- **Resolution**: Daily
+- **Data Density**: Dense
```
Comment (AlexCatarino) [inline listing-about.md:31]: `Dense` is reserved for tick resolution data. In this case, the density is "regular".

**Review by AlexCatarino — CHANGES_REQUESTED** [review]
Thank you for the contribution, @kuruma99.
Louis and I have minor comments mostly on using AddCrypto unnecessarily.

Additionally, I looks that has been a merge issue, please check out
[bnb.csv#L1693:L1700](
https://github.com/kuruma99/Lean.DataSource.CoinGeckoMarketCap/blob/master/output/alternative/coingecko/marketcap/bnb.csv#L1693:L1700)

**Pull request merged**